### PR TITLE
Use Strings instead of Vectors except when communicating

### PR DESCRIPTION
**Pull request by nicoleepp**
Closes issue #10

### CODE REVIEW

**Comment by iamed2** [comment]
I think we can wait for https://github.com/dask/distributed/pull/1328 to make this PR even better

**Comment by nicoleepp** [comment]
Yup that seems best

**Comment by nicoleepp** [comment]
Just rebased to resolve merge conflicts

**Comment by nicoleepp** [comment]
Tests won't pass until a new version of dask.distributed is released

**Comment by iamed2** [comment]
@nicoleepp new version of `dask.dispatcher` is `v0.18.1`

**Comment by nicoleepp** [comment]
@iamed2 I have rebased this branch now if you want to give it a final review

File: src/client.jl [hunk 1]
```diff
@@ -289,23 +278,13 @@ unnecessary computations.
 """
 get_key{T<:DispatchNode}(node::T) = error("$T does not implement get_key")
 
-function get_key(node::Op)
-    return string(
-        get_label(node), "-", hash((node.func, node.args, node.kwargs, node.result))
-    )
-end
+get_key(node::Op) = string(get_label(node), "-", hash((node)))
```
Comment (iamed2) [inline src/client.jl]: Extra parens in the `hash` calls here and below can be removed

File: src/worker.jl [hunk 2]
```diff
@@ -438,14 +438,15 @@ end
...
     communication
 """
 function get_data(worker::Worker; keys::Array=String[], who::String="")
     keys = Vector{String}(keys)
     debug(logger, "\"get_data\": ($keys: \"$who\")")
-    return Dict{Vector{UInt8}, Vector{UInt8}}(
-        Vector{UInt8}(k) =>
+
+    return Dict{String, Vector{UInt8}}(
+        k =>
```
Comment (iamed2) [inline src/worker.jl]: Put the pair on one line and if that line is too long put the `for` loop part on the next line, indented

File: src/client.jl [hunk 3]
```diff
@@ -317,84 +296,66 @@ Serialize dependencies to send to the scheduler.
...
-    client::Client,
-    node::DispatchNode
-)::Tuple{Dict{String, Vector{UInt8}}, Vector{DispatchNode}, Vector{Vector{UInt8}}}
-
-    # Get task dependencies
+function serialize_node(client::Client, node::DispatchNode)
     deps = collect(DispatchNode, dependencies(node))
     task = serialize_task(client, node, deps)
-    task_dependencies = Vector{UInt8}[Vector{UInt8}(get_key(dep)) for dep in deps]
+    task_dependencies =  map(get_key, deps)
```
Comment (iamed2) [inline src/client.jl]: Extra space before map can be removed

**Pull request merged**